packages/pdf-viewer/VerticalPages.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useLayoutEffect, useRef, useState } from 'react';
+import { PdfData, ScaledSize } from './pdfSource';
+import Page from './Page';
+
+require('./pages.css');
+
+export default function VerticalPages(props: {
+	pdf: PdfData;
+	isDarkTheme: boolean;
+	anchorPage: number;
+	container: React.MutableRefObject<HTMLElement>;
+}) {
+	const [scaledSize, setScaledSize] = useState<ScaledSize>(null);
+	const innerContainerEl = useRef<HTMLDivElement>(null);
+	const resizeTimer = useRef<number>();
+	useLayoutEffect(() => {
+		const updateSize = async () => {
+			if (innerContainerEl.current && props.pdf) {
+				const innerWidth = innerContainerEl.current.clientWidth;
+				const scaledSize = await props.pdf.getScaledSize(null, innerWidth - 10);
+				setScaledSize(scaledSize);
+			}
+		};
+		const onResize = () => {
+			if (resizeTimer.current) {
+				clearTimeout(resizeTimer.current);
+			}
+			resizeTimer.current = window.setTimeout(updateSize, 100);
+		};
+		window.addEventListener('resize', onResize);
+		updateSize()
+			.catch(console.error);
+		return () => {
+			window.removeEventListener('resize', onResize);
+			if (resizeTimer.current) {
+				clearTimeout(resizeTimer.current);
```

Comment:
resizeTimer is only used inside this effect, so define it here only. You don't need a ref

Reply:
Set the timer to null after clearing it